### PR TITLE
WK-178 update locations moved to the location reducer, fixed date picker

### PR DESCRIPTION
**Pull request by aditudorache**

### CODE REVIEW

File: src/components/map/Map.js [hunk 1]
```diff
@@ -23,26 +23,31 @@ const Map = () => {
...
-  React.useEffect(() => {
-    (async () => {
-      const blackspotsEndpoint = `${process.env.REACT_APP_API_ROOT}blackspots/spots/?format=geojson`;
-      fetchData(blackspotsEndpoint);
-    })();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => {
+    if (locations.length === 0)
+      (async () => {
+        const blackspotsEndpoint = `${process.env.REACT_APP_API_ROOT}blackspots/spots/?format=geojson`;
```
Comment (janjaap) [inline src/components/map/Map.js]: Wouldn't it be better to put the URL in a constants file? The previous PR had the URL written out in full

File: src/components/locationForm/LocationForm.jsx [hunk 2]
```diff
@@ -27,9 +26,14 @@ const LocationForm = ({ id }) => {
   const onSubmit = async data => {
     try {
       const url = `/api/blackspots/spots/${data.nummer}/`;
-      await sendData(url, toFormData(data), 'PATCH');
+      const location = await sendData(url, toFormData(data), 'PATCH');
+
+      const feature = toFeature(location);
+      actions.selectLocation({ payload: feature });
+      actions.updateLocation({ payload: feature });
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/components/locationForm/LocationForm.jsx]: sure!

File: src/components/locationForm/LocationForm.jsx [hunk 3]
```diff
@@ -27,9 +26,14 @@ const LocationForm = ({ id }) => {
...
     } catch (error) {
       // Dispatch the error message. console.log('Error! ', error);
+      console.log('Error! ', error);
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/components/locationForm/LocationForm.jsx]: will be removed later

File: src/components/map/Map.js (hunk 1 above)
Comment (aditudorache) [reply src/components/map/Map.js]: done.

File: src/components/locationForm/services/dateUtil.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ export const stringToDate = date => {
  * Temporary function that converts a date object in the dd/MM/yy format as expected in the api
  */
 export const dateToString = (date) => {
-  return date && `${date.getDate()}/${String(date.getMonth() +1).padStart(2, '0')}/${String(date.getFullYear()).substring(2)}`
+  return date && `${String(date.getDate()).padStart(2, '0')}/${String(date.getMonth() +1).padStart(2, '0')}/${String(date.getFullYear()).substring(2)}`
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/components/locationForm/services/dateUtil.js]: done

**Pull request merged**